packages/x-adapter/src/endpoint-adapter.factory.ts

```diff
@@ -37,10 +37,10 @@ export const endpointAdapterFactory: EndpointAdapterFactory = <Request, Response
     const endpoint = getEndpoint(rawEndpoint ?? requestEndpoint, request);
     const requestParameters = requestMapper(request, { endpoint });
 
-    return httpClient<Response>(
+    return httpClient(
       endpoint,
       deepMerge({}, defaultRequestOptions, requestOptions, { parameters: requestParameters })
-    ).then(response => responseMapper(response, { endpoint, requestParameters }));
+    ).then(response => responseMapper(response, { endpoint, requestParameters }) as Response);
```

Comment:
The issue we found yesterday comes from the type of the response mapper. As the `responseMapper` falls back to the `identityMapper`, the type is messed up. If you explicitly set the type of the destructured options this is fixed. I would fix it up there.

```ts
  const {
    endpoint: rawEndpoint,
    httpClient = fetchHttpClient,
    requestMapper = identityMapper,
    responseMapper = identityMapper,
    defaultRequestOptions = {}
  }: EndpointAdapterOptions<Request, Response> = options;
```